docs/concepts.md

```diff
@@ -48,7 +48,20 @@ A "region" is an entire interval of time (and not just a single point in time) t
 
 An entire recording has way too many objects that go in and out of existence and carry too many details for us to be able to hold all of that information in the memory at once. And that's why every recording is split into "regions" - smaller chunks of data that can be transmitted over network and stored in memory on the client, and further data derived from and interacted with. "Regions" are not something that user ever particularly cares about (beside being forced to wait for them to be loaded) and that they interact with, and instead, the user interacts with "pauses". A "pause" is that single moment in time for which the user can inspect the state of the application.
 
-A single `Pause` exists for each JS instruction. (TODO explain this and how it's related to debug stepping and `ExecutionPoint`s)
+A single `Pause` exists for each JS instruction. 
+
+#### How `Pause` is related to debug stepping and `ExecutionPoint`s
+
+In the Replay DevTools, a `Pause` represents the state of the JavaScript execution at a specific `ExecutionPoint`.Each `ExecutionPoint` corresponds to a specific line of code in the JavaScript program.
+
+When the Replay client requests that the backend pause the recording, it sends a request for a `Pause` object that contains information about the debuggee's state at the specific `ExecutionPoint`. This `Pause` object contains details about the `Frames`, `Scopes`, and `Objects` that were alive at that point in time.
+
+When the user interacts with the DevTools, they are stepping through the code and requesting new `Pause` objects for different `ExecutionPoints`. The user can step forward or backward through the `ExecutionPoints` to view the state of the JavaScript program at different points in time.
+
+Each `Pause` corresponds to a single `ExecutionPoint`, which means that there is one Pause for each line of code in the JavaScript program. This allows the user to debug the program by stepping through the code and viewing the state of the program at each line of code.
```

Comment:
This is not _entirely_ accurate. There may be no pauses, or there may be multiple pauses, for any given execution point.